After disabling JDBC config source, datasource properties remain required

I am using quarkus `3.1.1.Final` and quarkus config jdbc `2.2.0`.

I have added the JDBC config source like this:
```yaml
quarkus:
  config:
    source:
      jdbc:
        table: PARAMETER
        key: CODE
        value: VALUE_
```
I then add datasource properties to `.env` file:
```env
QUARKUS_DATASOURCE_JDBC_URL=jdbc:...
QUARKUS_DATASOURCE_USERNAME=user123
QUARKUS_DATASOURCE_PASSWORD=pass123
```
This works great, but for building the application in CI/CD pipeline I don't have a database and I don't need the properties to be fetched from the database. That's why I set the environment variable:
```env
QUARKUS_CONFIG_SOURCE_JDBC_ENABLED=false
```
However, the build fails with:
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.1.1.Final:build (default) on project maas-core: Failed to build quarkus application: Configuration validation failed:
[ERROR]         java.util.NoSuchElementException: SRCFG00011: Could not expand value quarkus.datasource.jdbc.url in property quarkus.config.source.jdbc.url
[ERROR]         java.util.NoSuchElementException: SRCFG00011: Could not expand value quarkus.datasource.password in property quarkus.config.source.jdbc.password
[ERROR]         java.util.NoSuchElementException: SRCFG00011: Could not expand value quarkus.datasource.username in property quarkus.config.source.jdbc.username
```
If I add the following environment variables, the build passes:
```env
QUARKUS_CONFIG_SOURCE_JDBC_URL=""
QUARKUS_CONFIG_SOURCE_JDBC_USERNAME=""
QUARKUS_CONFIG_SOURCE_JDBC_PASSWORD=""
```

Would it be possible to not expand the datasource properties, if `QUARKUS_CONFIG_SOURCE_JDBC_ENABLED` is set to `false` ?  You don't need those properties, if you don't use the JDBC config source.

--- a/jdbc/runtime/src/main/java/io/quarkiverse/config/jdbc/runtime/JdbcConfigConfig.java
+++ b/jdbc/runtime/src/main/java/io/quarkiverse/config/jdbc/runtime/JdbcConfigConfig.java
@@ -17,7 +17,7 @@ import io.smallrye.config.WithName;
  * The config itself is loaded using the ConfigSourceContext on the ConfigSourceFactory
  */
 @ConfigMapping(prefix = "quarkus.config.source.jdbc")
-@ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
 public interface JdbcConfigConfig {
 
     /**

--- a/jdbc/runtime/src/main/java/io/quarkiverse/config/jdbc/runtime/JdbcConfigSourceFactory.java
+++ b/jdbc/runtime/src/main/java/io/quarkiverse/config/jdbc/runtime/JdbcConfigSourceFactory.java
@@ -1,7 +1,9 @@
 package io.quarkiverse.config.jdbc.runtime;
 
 import java.sql.SQLException;
+import java.util.ArrayList;
 import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 
 import org.eclipse.microprofile.config.spi.ConfigSource;
@@ -9,13 +11,24 @@ import org.jboss.logging.Logger;
 
 import io.smallrye.config.ConfigSourceContext;
 import io.smallrye.config.ConfigSourceFactory;
+import io.smallrye.config.SmallRyeConfigBuilder;
 import io.smallrye.config.common.MapBackedConfigSource;
 
-public class JdbcConfigSourceFactory implements ConfigSourceFactory.ConfigurableConfigSourceFactory<JdbcConfigConfig> {
+public class JdbcConfigSourceFactory implements ConfigSourceFactory {
     private static final Logger log = Logger.getLogger(JdbcConfigSourceFactory.class);
 
+    // TODO - Cannot use ConfigurableConfigSourceFactory, because profiles are not being propagated. Need a fix in SR Config
     @Override
-    public Iterable<ConfigSource> getConfigSources(final ConfigSourceContext context, final JdbcConfigConfig config) {
+    public Iterable<ConfigSource> getConfigSources(final ConfigSourceContext context) {
+        List<String> profiles = new ArrayList<>(context.getProfiles());
+        Collections.reverse(profiles);
+
+        JdbcConfigConfig config = new SmallRyeConfigBuilder()
+                .withSources(new ConfigSourceContext.ConfigSourceContextConfigSource(context))
+                .withProfiles(profiles)
+                .withMapping(JdbcConfigConfig.class)
+                .build().getConfigMapping(JdbcConfigConfig.class);
+
         if (!config.enabled()) {
             return Collections.emptyList();
         }

